File: algorithms/1_easy/M/01827_minimum_operations_to_make_the_array_increasing/solution.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        count = 0
        curr_max = nums[0]

        for i in range(1, len(nums)):
            if nums[i] <= curr_max:
                curr_max += 1
                count += curr_max - nums[i]
            else:
                curr_max = nums[i]

        return count

File: algorithms/1_easy/M/01827_minimum_operations_to_make_the_array_increasing/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int count{0};
        int curr_max{nums[0]};

        for (auto it{nums.cbegin() + 1}; it != nums.cend(); ++it) {
            if (*it <= curr_max) {
                curr_max += 1;
                count += curr_max - *it;
            } else {
                curr_max = *it;
            }
        }

        return count;
    }
};
